pkg/internal/common/logging/metadata/level.go

```diff
@@ -0,0 +1,54 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package metadata
+
+import (
+	"sync"
+
+	"github.com/hyperledger/aries-framework-go/pkg/common/log"
+)
+
+const (
+	defaultLogLevel   = log.INFO
+	defaultModuleName = ""
+)
+
+//moduleLevels maintains log levels based on modules
+type moduleLevels struct {
+	mLevels map[string]log.Level
+	once    sync.Once
+}
+
+// GetLevel returns the log level for given module and level.
+func (l *moduleLevels) GetLevel(module string) log.Level {
+	level, exists := l.levels()[module]
+	if !exists {
+		level, exists = l.levels()[defaultModuleName]
+		// no configuration exists, default to info
+		if !exists {
+			return defaultLogLevel
+		}
+	}
+	return level
+}
+
+// SetLevel sets the log level for given module and level.
+func (l *moduleLevels) SetLevel(module string, level log.Level) {
+	l.levels()[module] = level
+}
+
+// IsEnabledFor will return true if logging is enabled for given module and level.
+func (l *moduleLevels) IsEnabledFor(module string, level log.Level) bool {
+	return level <= l.GetLevel(module)
+}
+
+func (l *moduleLevels) levels() map[string]log.Level {
+	l.once.Do(func() {
```

Comment:
Does the map have to be created lazily? You can create it when moduleLevels is created.

levels := &moduleLevels{mLevels: make(map[string]log.Level}

Reply:
It doesn't have to be lazy, will add NewXXX() functions and will remove sync.once.